exporter/src/main/scala/uk/gov/nationalarchives/consignmentexport/BagMetadata.scala

```diff
@@ -0,0 +1,102 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.domain.Metadata
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import org.keycloak.representations.idm.UserRepresentation
+import uk.gov.nationalarchives.consignmentexport.BagMetadata._
+import uk.gov.nationalarchives.consignmentexport.Config.Configuration
+
+class BagMetadata(graphQlApi: GraphQlApi, keycloakClient: KeycloakClient)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+  def getBagMetadata(consignmentId: UUID, config: Configuration): IO[Metadata] = for {
+   consignment <- graphQlApi.getConsignmentMetadata(config, consignmentId)
+   consignmentDetails = consignment match {
+     case Some(consignment) => getConsignmentDetails(consignment)
+     case None => throw new RuntimeException(s"No consignment metadata found for consignment $consignmentId")
+   }
+
+  } yield generateMetadata(consignmentDetails)
+
+  private def getConsignmentDetails(consignment: GetConsignment): Map[String, Option[String]] = {
+    val seriesCode = for {
+      series <- consignment.series
+      sc <- series.code
+    } yield sc
+
+    val bodyCode = for {
+      body <- consignment.transferringBody
+      bc <- body.code
+    } yield bc
+
+    val startDatetime = for {
+      createdDate <- consignment.createdDatetime
+      cd = createdDate.toString
+    } yield cd
+
+    val completedDatetime = for {
+      completedDate <- consignment.transferInitiatedDatetime
+      cd = completedDate.toString
+    } yield cd
+
+    val exportDatetime = for {
+      exportDate <- consignment.exportDatetime
+      ed = exportDate.toString
```

Comment:
I think we should do the same thing here as in the API, and be really explicit about the date-time format, rather than relying on it to do roughly the right thing. It will help the transformation service if we can be specific about the date format.